### PR TITLE
refactor: Merge task and controller for restoring seed phrase

### PR DESCRIPTION
**Pull request by kartikCypherock**
Implements https://app.clickup.com/t/9002019994/PRF-4797

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4797 Task controller merge - Wallet from seed phrase](https://app.clickup.com/t/9002019994/PRF-4797)

File: src/wallet/create_wallet_flow.h [hunk 1]
```diff
@@ -32,13 +31,10 @@
...
- * the X1 vault flash and X1 cards and verifies each share
- *
- * @param new_wallet true if a new wallet needs to be created, false if the user
- * wants to recover the wallet from a seed phrase
+ * @details This function generate random mnemonics, takes user inputs for
+ * wallet configuration, writes the wallet shares on the X1 vault flash and X1
+ * cards and verifies each share
  */
-void create_wallet_flow(bool new_wallet);
+void create_wallet_flow(void);
```
Comment (amanCypherock) [inline src/wallet/create_wallet_flow.h:38]: Some renaming could be done here, file and function both can be misleading now as the purpose has changed.

File: src/level_three/add_wallet/controller/controller_restore_wallet.c [hunk 2]
```diff
@@ -79,75 +77,186 @@ extern Flash_Wallet wallet_for_flash;
...
+    if (0xFFFF == choice) {
+      // This case arises when the user presses back button on the mnem ui
+
+      // If user presses back button on the first word, take them to the
+      // previous step.
+      if (0 == word) {
+        return INPUT_NUMBER_OF_MNEMONICS;
+      } else {
+        word -= 1;
+      }
```
[the remark this answers is not in the document]
Comment (kartikCypherock) [reply src/level_three/add_wallet/controller/controller_restore_wallet.c:138]: The checks will still remain the same.. would it not? Can you explain what is the exact benefit we are looking for here

File: src/wallet/create_wallet_flow.h (hunk 1 above)
Comment (kartikCypherock) [reply src/wallet/create_wallet_flow.h]: Makes sense, updated

File: src/level_three/add_wallet/controller/controller_restore_wallet.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (kartikCypherock) [reply src/level_three/add_wallet/controller/controller_restore_wallet.c:138]: We should discuss the strategy for state management and then resolve issues of such kind.

File: src/wallet/create_wallet_flow.h (hunk 1 above)
Comment (kartikCypherock) [reply src/wallet/create_wallet_flow.h]: Updated the name

File: src/level_three/add_wallet/controller/controller_restore_wallet.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (amanCypherock) [reply src/level_three/add_wallet/controller/controller_restore_wallet.c:138]: As we discussed, both implementations work but have different expectations from `get_state_on_list_scr`. We can resolve this now as there's no performance or function issue with the implementation.

**Review by amanCypherock — APPROVED** [review]

**Pull request merged**